### PR TITLE
Change dmi-api reference method

### PR DESCRIPTION
**Pull request by wbc6080**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Previously, mapper-framework copied the dmi-api of kubeedge to /pkg for reference. It was not easy to modify when the api was updated.
The current PR directly references kubeedge in go.mod and references dmi-api in the form of a  package, which can automatically synchronize the modification of the main library api.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by ZhengXinwei-F** [comment]
lgtm

**Comment by wbc6080** [comment]
PTAL @cl2017  @WillardHu

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/devicetwin.go [hunk 1]
```diff
@@ -11,9 +11,9 @@ import (
 
 	"github.com/kubeedge/Template/driver"
 	"github.com/kubeedge/Template/pkg/common"
-	dmiapi "github.com/kubeedge/Template/pkg/dmi-api"
```
[the remark this answers is not in the document]
Comment (wbc6080) [reply staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/devicetwin.go:14]: How about first modify the way of referencing dmi-api, waiting for a while to see whether there is other problem. If not, then we can delete them @cl2017

[the remark this answers is not in the document]
Comment (cl2017) [reply staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/devicetwin.go:14]: ok, no problem~

**Comment by wbc6080** [comment]
PTAL @WillardHu , thanks

**Comment by WillardHu** [comment]
lgtm

File: staging/src/github.com/kubeedge/mapper-framework/go.mod [hunk 2]
```diff
@@ -5,19 +5,29 @@ go 1.20
...
 )
+
+replace (
+	github.com/kubeedge/beehive v0.0.0 => github.com/kubeedge/beehive v1.13.0
+	github.com/kubeedge/viaduct v0.0.0 => github.com/kubeedge/viaduct v1.13.0
+	k8s.io/component-helpers => github.com/kubeedge/kubernetes/staging/src/k8s.io/component-helpers v1.26.10-kubeedge1
+	k8s.io/dynamic-resource-allocation => github.com/kubeedge/kubernetes/staging/src/k8s.io/dynamic-resource-allocation v1.26.10-kubeedge1
+	k8s.io/legacy-cloud-providers => github.com/kubeedge/kubernetes/staging/src/k8s.io/legacy-cloud-providers v1.26.10-kubeedge1
+	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.26.10
+)
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/mapper-framework/go.mod:33]: We can set the go.mod as below:
```
module github.com/kubeedge/Template
  
go 1.20

require (
        github.com/kubeedge/kubeedge v1.15.1
)

```
Then run `hack/verify-vendor.sh` to update the gomod, gosum and vendor related files. L26-33 will be auto deleted.

Comment (fisherxu) [reply staging/src/github.com/kubeedge/mapper-framework/go.mod:33]: And I think we can create a new repo for KubeEdge API, so mapper and other clients can import the api, not all kubeedge repo, how do you think? @WillardHu @cl2017 @Shelley-BaoYue

Comment (WillardHu) [reply staging/src/github.com/kubeedge/mapper-framework/go.mod:33]: > And I think we can create a new repo for KubeEdge API, so mapper and other clients can import the api, not all kubeedge repo, how do you think? @WillardHu @cl2017 @Shelley-BaoYue

 STGM; +1

Comment (cl2017) [reply staging/src/github.com/kubeedge/mapper-framework/go.mod:33]: > And I think we can create a new repo for KubeEdge API, so mapper and other clients can import the api, not all kubeedge repo, how do you think? @WillardHu @cl2017 @Shelley-BaoYue

Yes, very good idea. +1

Comment (wbc6080) [reply staging/src/github.com/kubeedge/mapper-framework/go.mod]: OK, I have fixed it @fisherxu

Comment (fisherxu) [reply staging/src/github.com/kubeedge/mapper-framework/go.mod]: > And I think we can create a new repo for KubeEdge API, so mapper and other clients can import the api, not all kubeedge repo, how do you think? @WillardHu @cl2017 @Shelley-BaoYue

Okey, @wbc6080 please open an issue to track the api repo things, and let's discuss whether should we place the api dir to staging.

**Review by fisherxu — COMMENTED** [review]
/lgtm
/assign @cl2017 @WillardHu

**Comment by WillardHu** [comment]
lgtm

**Comment by wbc6080** [comment]
I have opened it in #5301

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**